include/villas/config_helper.hpp

```diff
@@ -42,5 +46,102 @@ int json_object_extend(json_t *orig, json_t *merge);
 
 json_t *json_load_cli(int argc, const char *argv[]);
 
+template <typename Duration = std::chrono::nanoseconds>
+Duration parse_duration(const std::string &input) {
+  using namespace std::literals::chrono_literals;
+
+  // Map unit strings to their corresponding chrono durations
+  static const std::unordered_map<std::string, std::chrono::nanoseconds>
+      unit_map = {
+          {"d", 24h},  // days
+          {"h", 1h},   // hours
+          {"m", 1min}, // minutes
+          {"s", 1s},   // seconds
+          {"ms", 1ms}, // milliseconds
+          {"us", 1us}, // microseconds
+          {"ns", 1ns}  // nanoseconds
+      };
+
+  std::regex token_re(R"((\d+)([a-z]+))", std::regex::icase);
```

Comment:
Is it really necessary to do case-insensitive matching? I am not convinced that leniency in parsing does us any good here.

E.g. should we really accept `Ms` as milliseconds?

Reply:
I agree. Done.